### ISSUE TITLE
**Important (Timestep Issue)

### ISSUE THREAD

**Issue by DriveHabits** [bmaltais/kohya_ss#2797]
I noticed something quite odd but it seemed to fix my issue. I was doing style training.

1. You had a typo for flux shift in the timestep sampling on a previous commit that you then fixed.

2. I did a training when it had that issue and I noticed my style was working. I'm not sure what the behavior of having that typo in, idk if flux was even timestep sampling at all, but I had better results with that typo.

Now for the question, is it possible to implement and option where we can disable timestep sampling or having something similar to the typo version?

Thank you!

**Comment by DarkViewAI** [comment]
I actually noticed this as well. Seems the typo version had better results

**Comment by bmaltais** [comment]
Make sure to use the same seed for the training. This would help maintain training results consistency. The training seed can have a similar effect on the end result as the seed has on prompt outputs.

**Comment by DriveHabits** [comment]
> Make sure to use the same seed for the training. This would help maintain training results consistency. The training seed can have a similar effect on the end result as the seed has on prompt outputs.

Yep i made sure all was the same, i've ran 40+ trainings to confirm

**Comment by DriveHabits** [comment]
i used training seed of 42 for all. I wonder what the typo did differently that made the results better

**Comment by DriveHabits** [comment]
<img width="1100" alt="cxcx" src="https://github.com/user-attachments/assets/142cb13c-44c6-49a4-b320-b331bb23a88e">
loaded up the working config, in the toml it shows up as the typo "flux_shift "

tried a training on it, works great.

then tried one with the actual fixed commit and training results are not even close.

**Comment by bmaltais** [comment]
I believe the default value is the one used when the one specified is not valid… So it probably was defaulting to “shift” instead.

**Comment by bmaltais** [comment]
Actually, the default is “sigma”

**Comment by DriveHabits** [comment]
@bmaltais even when its passed through the toml? it defaults to sigma?

**Comment by bmaltais** [comment]
I suspect so

**Comment by DriveHabits** [comment]
> I suspect so

@bmaltais i tried sigma with the same settings, results are quite bad

**Comment by bmaltais** [comment]
Him… this is strange as the sd-scripts will use that if not specified… but maybe if a “bad” value is specified it a toml config it default to something else? Maybe @kohya-ss could tell us what happens in that case.

**Comment by DriveHabits** [comment]
@bmaltais here is the config.toml with the typo it had 
https://huggingface.co/DriveHabits/Test/blob/main/config_lora-20240901-142242.toml

**Comment by DarkViewAI** [comment]
> If `flux_shift ` (with an extra space) is specified, it will be `sigma`. If you train with `sigma`, you should get the same condition.
> 
> If something is different with the same seed, it may be due to other script changes, for example `--network_train_unet_only` now works.

and what about the discrete flow shift? when i used the flux shift with the extra space, i used 3.1582? would i still use the same or does sigma use something else?

**Comment by DarkViewAI** [comment]
@kohya-ss thanks

**Comment by DarkViewAI** [comment]
> > and what about the discrete flow shift? when i used the flux shift with the extra space, i used 3.1582? would i still use the same or does sigma use something else?
> 
> Discrete flow shift is ignored when `timestep_sampling` is other than `shift`. So it is ignored in `sigma` or `flow_shift`.

thank you will try it out and see if I get something similar with sigma
